### PR TITLE
BBOB module

### PR DESCRIPTION
**Pull request by SunilMahendrakar**
Added BBOB benchmarking.

### CODE REVIEW

**Comment by SunilMahendrakar** [comment]
Travis fails due to boost/filesystem not being found when importing in python. @kiryx can you please take a look.

File: src/util/bbob.cpp [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+
+                //try to find "funcId = "
+                std::string pattern("funcId = ");
+                std::size_t position = line.find(pattern);
+                if(position != std::string::npos) //required line
+                {
+                    found = 0;
+                    boost::split(tokens, line, boost::is_any_of(","));
+
+                    for(int i = 0; i < tokens.size(); i++)
```
Comment (krzysztof) [inline src/util/bbob.cpp]: signed and unsigned int comparison

File: src/util/bbob.h [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        const unsigned int m_nbPtsEvals = 20;
+        const unsigned int m_nbPtsF = 5;
+
+        mutable LastEvalStruct m_LastEval, m_BestFEval;
+        mutable double m_lastWriteEval;
+
+        std::string m_algName; //Name of the optimiser (used in the post-processing)
+        std::string m_comments; //Additional comments
+        double m_precision = 1e-8; //The precision
+        unsigned int m_instanceId = 1; //The instance ID
```
Comment (krzysztof) [inline src/util/bbob.h]: -Wreorder:
put `m_instanceId` before `m_comments`

Comment (SunilMahendrakar) [inline src/util/bbob.h]: I reordered in the constructor.

**Comment by SunilMahendrakar** [comment]
closing this. Merged in https://github.com/kiryx/pagmo/pull/6

**Pull request closed without merging**